logstash-core/lib/logstash/plugin.rb

```diff
@@ -99,6 +107,13 @@ def metric
     @metric_plugin ||= enable_metric ? @metric : LogStash::Instrument::NullMetric.new
   end
 
+  # return the configured name of this plugin
+  # @return [String] The name of the plugin defined by `config_name`
+  def config_name
```

Comment:
As this is only used in the `plugin_unique_name` method, can we just there just the class method call? what do you think?